### PR TITLE
Usage Insights: Make it toggleable

### PR DESCRIPTION
**Pull request by joanlopez**
**What is this feature?**

These changes contribute to make Usage Insights service toggleable (make it possible to enable/disable it).

**Why do we need this feature?**

In short: in certain cases, it makes no sense to keep Usage Insights enabled, but you can find further details [here](https://github.com/grafana/grafana-enterprise/issues/4164).

**Who is this feature for?**

Grafana administrators / operators

**Which issue(s) does this PR fix?**:

Contributes to https://github.com/grafana/grafana-enterprise/issues/4164

**Special notes for your reviewer**:

Tested and seems to be worked as expected, but any extra test will be more than welcome.

Complement to https://github.com/grafana/grafana-enterprise/pull/4165

### CODE REVIEW

File: packages/grafana-runtime/src/config.ts [hunk 1]
```diff
@@ -141,7 +141,7 @@ export class GrafanaBootConfig implements GrafanaConfig {
   rudderstackDataPlaneUrl: undefined;
   rudderstackSdkUrl: undefined;
   rudderstackConfigUrl: undefined;
-
+  usageInsightsEnabled =  true;
```
Comment (joanlopez) [inline packages/grafana-runtime/src/config.ts:144]: I guess this is safe even for those cases when the feature is disabled in Grafana settings, isn't it? 
Or should I defined it `undefined` by default? 🤔

File: packages/grafana-runtime/src/config.ts [hunk 2]
```diff
@@ -122,7 +122,7 @@ export class GrafanaBootConfig implements GrafanaConfig {
   geomapDefaultBaseLayerConfig?: MapLayerOptions;
   geomapDisableCustomBaseLayer?: boolean;
   unifiedAlertingEnabled = false;
-  unifiedAlerting = { minInterval: '' };
+  unifiedAlerting = {minInterval: ''};
```
Comment (joanlopez) [inline packages/grafana-runtime/src/config.ts:125]:
```suggestion
  unifiedAlerting = { minInterval: '' };
```

File: packages/grafana-runtime/src/config.ts [hunk 3]
```diff
@@ -189,11 +189,11 @@ export class GrafanaBootConfig implements GrafanaConfig {
 function getThemeCustomizations(config: GrafanaBootConfig) {
   const mode = config.bootData.user.lightTheme ? 'light' : 'dark';
   const themeOptions: NewThemeOptions = {
-    colors: { mode },
+    colors: {mode},
   };
 
   if (config.featureToggles.interFont) {
-    themeOptions.typography = { fontFamily: '"Inter", "Helvetica", "Arial", sans-serif' };
+    themeOptions.typography = {fontFamily: '"Inter", "Helvetica", "Arial", sans-serif'};
```
Comment (joanlopez) [inline packages/grafana-runtime/src/config.ts:196]:
```suggestion
    themeOptions.typography = { fontFamily: '"Inter", "Helvetica", "Arial", sans-serif' };
```

File: pkg/setting/setting.go [hunk 4]
```diff
@@ -179,6 +179,9 @@ var (
 	GrafanaComUrl string
 
 	ImageUploadProvider string
+
+	// Usage Insights
+	UsageInsightsEnabled bool
```
Comment (joanlopez) [inline pkg/setting/setting.go:184]: Not ideal, but looked (imho) safer than reading it directly from settings (e.g. Settings Provider) and "duplicating" or "re-defining" its default value. So, I defined it here to let the owner service (MetaAnalytics) to set the corresponding value.

**Comment by Clarity-89** [comment]
The feature toggles are now generated, the instructions are in `toggles_gen.go` for example:
```
// To change feature flags, edit:
//  pkg/services/featuremgmt/registry.go
// Then run tests in:
//  pkg/services/featuremgmt/toggles_gen_test.go
```

This will generate both front end and backend constants for the toggles.

**Comment by joanlopez** [comment]
Closed in favor of https://github.com/grafana/grafana-enterprise/pull/4203

**Pull request closed without merging**